22.md

```diff
@@ -0,0 +1,70 @@
+LUD-22: Payee identity data in `payRequest` protocol.
+================================================
+
+`author: onthedeklein` `discussion: https://t.me/lnurl/36488`
+
+---
+
+This is the payee-side equivalent of [LUD-18](18.md). It gives the opportunity for the payer to request that the payee provide identitifying information before the payment is made. This allows the sending `WALLET` to show information about the receiver in order to allow the payer to verify that they are paying the correct entity. This also gives the opportunity for the payee to authorize themselves via a challenge-response mechanism to provide some assurance that `SERVICE` has not been compromised.
+
+## 1. `payeeData` record
+
+If `WALLET` wants to get one or more types of payee identities/data from `SERVICE` then it MUST alter its JSON request to the second callback to include a `payeeData` query param, as follows (notice that the `payeeData` record below has a bunch of fields only for completion, an actual response will likely contain just a subset of these):
+
+```typescript
+ {
+  "name": { "mandatory": boolean },
+  "pubkey": { "mandatory": boolean },
+  "identifier": { "mandatory": boolean },
+  "email": { "mandatory": boolean },
+  "auth": {
+     "mandatory": boolean,
+     "k1": string // hex encoded 32 bytes of challenge
```

Comment:
Considering changing the name k1 here to avoid confusion with LUD-04, since it's serving a slightly different purpose. Seeking opinions.